sound/soc/sof/pcm.c

```diff
@@ -168,9 +168,6 @@ static int sof_pcm_hw_params(struct snd_pcm_substream *substream,
 	/* save pcm hw_params */
 	memcpy(&spcm->params[substream->stream], params, sizeof(*params));
 
-	/* unset restore_stream */
-	spcm->restore_stream[substream->stream] = 0;
-
```

Comment:
why not leave this here?

Reply:
@plbossart I cant clear it here because I also use the same flag for setting the hw_params for the BE dai link which happens later. Clearing it here will prevent the BE hw_params from being setup after resuming.

The reason I reset the flag in open is because we can be sure that when the stream is opened, it will be followed by hw_params and prepare(). And, we do not want to repeat it in both these calls.

Reply:
ok, I will have to trust you on this one.